subsys/bluetooth/host/hci_core.c

```diff
@@ -3413,6 +3417,22 @@ void hci_event_prio(struct net_buf *buf)
 	}
 }
 
+#if !defined(CONFIG_BT_RECV_BLOCKING)
+static void rx_queue_put(struct net_buf *buf)
+{
+	net_buf_put(&bt_dev.rx_queue, buf);
+
+#if defined(CONFIG_BT_RECV_WORKQ_SYS)
+	int err = k_work_submit(&rx_work);
+#elif defined(CONFIG_BT_RECV_WORKQ_BT)
+	int err = k_work_submit_to_queue(&bt_workq, &rx_work);
+#endif /* CONFIG_BT_RECV_WORKQ_SYS */
+	if (err < 0) {
+		BT_ERR("Could not submit rx_work: %d", err);
+	}
+}
+#endif /* !CONFIG_BT_RECV_BLOCKING */
```

Comment:
This one is a bit fun, because the function itself is guarded with `!defined(CONFIG_BT_RECV_BLOCKING)` and the individual work items are guarded with `CONFIG_BT_RECV_WORKQ_SYS` and `CONFIG_BT_RECV_WORKQ_BT`. 

So in the case that we add another WORKQ option, this function will start compile properly, but won't work as intended. 

I would suggest to guard the function with `#if defined(CONFIG_BT_RECV_WORKQ_SYS) || defined(CONFIG_BT_RECV_WORKQ_BT)` instead, to ensure that it isn't called without support, and also commonlize a bit more:

```suggestion
#if defined(CONFIG_BT_RECV_WORKQ_SYS) || defined(CONFIG_BT_RECV_WORKQ_BT)
static void rx_queue_put(struct net_buf *buf)
{
	net_buf_put(&bt_dev.rx_queue, buf);

#if defined(CONFIG_BT_RECV_WORKQ_SYS)
	const int err = k_work_submit(&rx_work);
#elif defined(CONFIG_BT_RECV_WORKQ_BT)
	const int err = k_work_submit_to_queue(&bt_workq, &rx_work);
#endif /* CONFIG_BT_RECV_WORKQ_SYS */

	if (err < 0) {
		BT_ERR("Could not submit rx_work: %d", err);
	}
}
#endif /* CONFIG_BT_RECV_WORKQ_SYS || CONFIG_BT_RECV_WORKQ_BT */
```